File: JavaScript/subscriber.js
const mqtt = require('mqtt')

const broker = "mqtt://test.mosquitto.org"
const topico = 'siChat/mqtt'

// Conectar ao broker
const client = mqtt.connect(broker)

client.on('connect', () => {
    console.log('Conectado ao broker')

    //Inscrever-se no tópico
    client.subscribe(topico, (err) => {
        if(!err){
            console.log(`Inscrito no tópico: ${topico}`)
        }
    })
})

client.on('message', (topico, message) => {
    //Exibir a mensagem recebida
    console.log(`Nova mensagem no tópico ${topico}: ${message.toString()}`)
})

client.on('error', (err) =>{
    console.log('Erro de conexão: ', err)
    client.end()
})

File: Python/subscriber.py
import paho.mqtt.client as mqtt

# Configurações do Broker
broker = "test.mosquitto.org"
porta = 1883
topico = "siChat/mqtt"

# Função que será chamada quando uma nova mensagem for recebida
def on_menssage(client, userData, message):
    print(f"Nova mensagem recebida: {message.payload.decode('utf-8')}")

# Função para quando a conexão for estabelecida com o broker
def on_connect(client, userData, flags, rc):
    print(f"Conectado ao Broker com código de resultado {rc}")
    # Inscrever-se no tópico
    client.subscribe(topico)

# Criar cliente MQTT
client = mqtt.Client()

# Definir callback de conexão
client.on_connect = on_connect

# Definir callback para recebimento de mensagens
client.on_message = on_menssage

# Conectar ao broker
client.connect(broker, porta, 60)

# Iniciar o loop de rede para processar mensagens
client.loop_forever()